core/src/main/java/org/infinispan/persistence/sifs/Compactor.java

```diff
@@ -272,10 +299,18 @@ private void resumeAfterPause() {
    }
 
    public void stopOperations() {
+      // This will short circuit any compactor call, so it can only process the entry it may be on currently
       terminateSignal = true;
       processor.onComplete();
-      Util.close(logFile);
-      logFile = null;
+      // The stopped CompletableFuture is completed in onComplete or onError callback for the processor, so this will
+      // return after all compaction calls are completed
+      stopped.join();
+      if (logFile != null) {
+         Util.close(logFile);
+         // Complete the file, this file should not be compacted
+         completeFile(logFile.fileId, currentOffset, nextExpirationTime);
```

Comment:
Nitpicking here as it doesn't actually affect the execution, but a `false` at the end to not even invoke `schedule`?

Reply:
Sure, it would be more explicit then.